web/ui/react-app/src/components/CustomInfiniteScroll.tsx

```diff
@@ -0,0 +1,50 @@
+import { ComponentType, useEffect, useState } from 'react';
+import InfiniteScroll from 'react-infinite-scroll-component';
+
+const initialNumberOfTargetsDisplayed = 50;
+
+export interface InfiniteScrollItemsProps<T> {
+  items: T[];
+}
+
+interface InfiniteScrollProps<T> {
+  value: T[];
+  child: ComponentType<InfiniteScrollItemsProps<T>>;
+}
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+const CustomInfiniteScroll = <T extends unknown>({ value, child }: InfiniteScrollProps<T>) => {
+  const [items, setItems] = useState<T[]>(value.slice(0, 50));
+  const [index, setIndex] = useState<number>(initialNumberOfTargetsDisplayed);
+  const [hasMore, setHasMore] = useState<boolean>(value.length > initialNumberOfTargetsDisplayed);
+  const Child = child;
```

Comment:
Hmm, wondering if there's a more elegant way for handling single, parametrized child components, but not sure.